config/lib/express.js

```diff
@@ -40,8 +40,9 @@ module.exports.initLocalVariables = function (app) {
 
   // Passing the request url to environment locals
   app.use(function (req, res, next) {
-    res.locals.host = req.protocol + '://' + req.hostname;
-    res.locals.url = req.protocol + '://' + req.headers.host + req.originalUrl;
+    var domain = config.app.domain || req.headers.hostname;
```

Comment:
Using req.hostname is dangerous, since that enables HTTP Host Header attacks. It doesn't really matter for livereload, but the variable `res.locals.host` could have been used for more important things as well.

Using the port from the environment breaks when the app runs behind a proxy (ie on Heroku or AWS). In this case the port of the app is completely arbitary while the external port is still 80.